### PR TITLE
Fix bug in h5_table_to_astropy and HDF5TableWriter

### PR DESCRIPTION
**Pull request by kosack**
## Expected behavior:
```
table = h5_table_to_astropy(dl1_file, path_to_dataset)
table.write("example.ecsv")
```

This currently fails due to several reasons:
1. the _DESC and other metadata were stored as numpy strings, not python strings, causing astropy's io mechanism to fail with unknown type
2. the _DESC metadata were erroneously written for skipped columns in HDF5TableWriter 
3. the _DESC data should be used as a table.column[x].description attribute

## Fix:
- don't write _DESC for skipped columns
- convert all attribute metadata  to strings `str(value)` (or should we check for only things that were strings?) 
- use the .description attribute of Column class in astropy

The output now looks like
image

### CODE REVIEW

**Comment by maxnoe** [comment]
> convert all attribute metadata to strings str(value) (or should we check for only things that were strings?)

Yes, please only convert `np.str_` to string. Since ints and floats and all other dtypes are usable here.

File: ctapipe/io/astropy_helpers.py [hunk 1]
```diff
@@ -62,7 +62,7 @@ def h5_table_to_astropy(h5file, path) -> QTable:
             # need to convert to str() here so they are python strings, not
             # numpy strings
             value = table.attrs[attr]
-            other_attrs[attr] = str(value) if isinstance(value, np.str) else value
+            other_attrs[attr] = str(value) if isinstance(value, np.str_) else value
```
Comment (maxnoe) [inline ctapipe/io/astropy_helpers.py:65]:
```
>>> issubclass(np.str_, np.str)
True
```
So using `np.str` is probably more general.

**Review by maxnoe — DISMISSED** [review]

File: ctapipe/io/astropy_helpers.py (hunk 1 above)
Comment (kosack) [reply ctapipe/io/astropy_helpers.py:65]: yes, I had that originally (see the commit message) and it worked, but what is the point of converting a str → str if it is not a numpy str_?  Doesn't hurt, but doesn't do anything either.

Comment (kosack) [reply ctapipe/io/astropy_helpers.py:65]: unless we expect other string types, perhaps

**Review by watsonjj — DISMISSED** [review]

File: ctapipe/io/astropy_helpers.py (hunk 1 above)
Comment (maxnoe) [reply ctapipe/io/astropy_helpers.py:65]: Ah, sorry, didn't realize that `np.str` is the python string. Sorry. That's confusing.

**Comment by kosack** [comment]
sorry @watsonjj and @maxnoe , please re-review this.  The tests now work again (they broke due to a typo being accidentally committed)

**Review by maxnoe — APPROVED** [review]

**Comment by kosack** [comment]
this is ready, other than perhaps changing the name to `ctapipe.io.read_table` or something simpler than `ctapipe.io.astropy_helpers.h5_table_to_astropy`

**Review by maxnoe — APPROVED** [review]

**Review by vuillaut — APPROVED** [review]

**Review by maxnoe — APPROVED** [review]

**Pull request merged**